File: tasks_api/src/main/java/com/brenodev/tasks/model/Task.java
package com.brenodev.tasks.model;

import com.brenodev.tasks.model.enums.Priority;
import com.brenodev.tasks.model.enums.Status;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "tb_tasks")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Priority priority;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @PrePersist
    protected void onCreate(){
        this.createdAt = LocalDateTime.now();
        if (this.status == null) this.status = Status.PENDING;
    }
}

File: task_app/lib/models/task.dart
class Task {
  final int? id;
  final String title;
  final String description;
  final String priority;
  final String status;
  final int userId;
  final String? createdAt;

  Task({
    this.id,
    required this.title,
    required this.description,
    required this.priority,
    this.status = 'PENDING',
    required this.userId,
    this.createdAt,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      description: json['description'] ?? '',
      priority: json['priority'],
      status: json['status'],
      userId: json['user']['id'],
      createdAt: json['createdAt'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'priority': priority,
      'status': status,
      'user': {'id': userId},
    };
  }

  Map<String, dynamic> toCreateJson() {
    return {
      'title': title,
      'description': description,
      'priority': priority,
      'user': {'id': userId},
    };
  }

  Task copyWith({String? status, String? createdAt}) {
    return Task(
      id: id,
      title: title,
      description: description,
      priority: priority,
      status: status ?? this.status,
      userId: userId,
      createdAt: createdAt ?? this.createdAt,
    );
  }

  bool get isCompleted => status == 'COMPLETED';
}
